subprojects/gst-plugins-good/sys/v4l2/gstv4l2allocator.c

```diff
@@ -357,7 +357,11 @@ gst_v4l2_allocator_release (GstV4l2Allocator * allocator, GstV4l2Memory * mem)
 
   switch (allocator->memory) {
     case V4L2_MEMORY_DMABUF:
+      if (V4L2_TYPE_IS_CAPTURE(allocator->obj->type) && (allocator->obj->mode == GST_V4L2_IO_DMABUF_IMPORT))
+        break;
```

Comment:
Please, consider backporting the upstream patch fixing this instead of deviating. This type of change would make future GStreamer upgrade difficult.

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5101

Reply:
Modified as per the suggestion

Reply:
By backporting, I meant to port the patch "v4l2: allocator: Don't close foreign dmabuf", and apply it as its own patch. This way we know where the change is from as it has cross-reference and original authorship.

Reply:
Modified as per your suggestion. Please review and comment.